src/madness/world/worldrmi.cc

```diff
@@ -401,56 +399,20 @@ namespace madness {
             }
 
             MADNESS_ASSERT(task_ptr == nullptr);
-#if HAVE_INTEL_TBB
-
-            // Force the RMI task to be picked up by someone other than main thread
-            // by keeping main thread occupied AND enqueing enough dummy tasks to make
-            // TBB create threads and pick up RmiTask eventually
+            task_ptr.reset(new RmiTask(comm));
 
-            tbb_rmi_parent_task =
-                new (tbb::task::allocate_root()) tbb::empty_task;
-            tbb_rmi_parent_task->set_ref_count(2);
-            task_ptr = new (tbb_rmi_parent_task->allocate_child()) RmiTask(comm);
-#ifdef MADNESS_CAN_USE_TBB_PRIORITY
-            tbb::task::enqueue(*task_ptr, tbb::priority_high);
-#else
-            tbb::task::enqueue(*task_ptr);
-#endif  // MADNESS_CAN_USE_TBB_PRIORITY
+#if HAVE_INTEL_TBB
+            ThreadPool::tbb_arena->enqueue([]{
+                task_ptr->run();
+            });
 
+            //TODO: is it needed ?
             task_ptr->comm.Barrier();
 
-            // repeatedly hit TBB with binges of empty tasks to force it create threads
-            // and pick up RmiTask;
-            // paranoidal sanity-check against too many attempts
-            tbb::task* empty_root =
-                new (tbb::task::allocate_root()) tbb::empty_task;
-            empty_root->set_ref_count(1);
-            int binge_counter = 0;
-            const int max_binges = 10;
             while (!rmi_task_is_running) {
-              if (binge_counter == max_binges)
-                throw madness::MadnessException("MADWorld failed to launch RMI task",
-                                                "ntries > 10",
-                                                10,__LINE__,__FUNCTION__,__FILE__);
-              const int NEMPTY = 1000000;
-              empty_root->add_ref_count(NEMPTY);
-              for (int i = 0; i < NEMPTY; i++) {
-                tbb::task* empty =
-                    new (empty_root->allocate_child()) tbb::empty_task;
-#ifdef MADNESS_CAN_USE_TBB_PRIORITY
-                tbb::task::enqueue(*empty, tbb::priority_high);
-#else
-                tbb::task::enqueue(*empty);
-#endif  // MADNESS_CAN_USE_TBB_PRIORITY
-                ++binge_counter;
-              }
               myusleep(100000);
             }
-            empty_root->wait_for_all();
-            tbb::task::destroy(*empty_root);
-            task_ptr->comm.Barrier();
```

Comment:
What about if worker threads are not available? Previously, `RmiTask` can be picked up with the current thread.